components/Common/Tabs/index.tsx

```diff
@@ -0,0 +1,37 @@
+import * as TabsPrimitive from '@radix-ui/react-tabs';
+import type { FC, ReactNode } from 'react';
+
+import styles from './index.module.css';
+
+type Tab = {
+  name: string;
+  label: ReactNode;
+  content: ReactNode;
+};
+
+type TabsProps = {
+  tabs: Tab[];
+} & TabsPrimitive.TabsProps;
+
+const Tabs: FC<TabsProps> = ({ tabs, ...props }) => (
+  <TabsPrimitive.Root {...props}>
+    <TabsPrimitive.List className={styles.tabsList}>
+      {tabs.map(tab => (
+        <TabsPrimitive.Trigger
+          key={tab.name}
+          value={tab.name}
+          className={styles.tabsTrigger}
+        >
+          {tab.label}
+        </TabsPrimitive.Trigger>
+      ))}
+    </TabsPrimitive.List>
+    {tabs.map(tab => (
```

Comment:
Let's do the following, since, the TabsPrimitive.Content doesn't have a base styling, let's remove the `map` from here.

Let's update the type of Tab to just be:

```tsx
type Tab = {
  key: string;
  label: ReactNode;
};
```

And then the type of `TabsProps`:

```tsx
type TabsProps = {
  tabs: Tab[];
  headerClassName?: string; // optional extra class name for the tabsprimitive.list;
} & TabPrimitive.TabsProps;
```

And then the `Tabs` component is. `FC<PropsWithChildren<TabsProps>>`

Meaning, the children is `{children}` and inside the children we can eventually add the Tabs Content, so we don't need to re-export `TabsPrimitive.Content` nor map them